packages/lesswrong/lib/collections/users/helpers.ts

```diff
@@ -295,45 +316,60 @@ interface UserLocation {
-    // geolocation API. This is not always available.
-    if (typeof window !== 'undefined' && typeof navigator !== 'undefined'
-        && navigator && navigator.geolocation) {
+
+  const [location, setLocation] = useState(() => {
+    if (currentUserLat && currentUserLng) {
+      // First return a location from the user profile, if set
+      return {lat: currentUserLat, lng: currentUserLng, loading: false, known: true}
+    } else if (isServer) {
+      // If there's no location in the user profile, we may still be able to get
+      // a location from the browser--but not in SSR.
+      return {lat: placeholderLat, lng: placeholderLng, loading: true, known: false}
+    } else {
+      // If we're on the browser, we'll try to get a location using the browser
+      // geolocation API. This is not always available.
+      if (typeof window !== 'undefined' && typeof navigator !== 'undefined' && navigator && navigator.geolocation) {
+        return {lat: placeholderLat, lng: placeholderLng, loading: true, known: false}
+      }
+    }
+  
+    return {lat: placeholderLat, lng: placeholderLng, loading: false, known: false}
+  })
+  
+  useEffect(() => {
+    // if we don't yet have a location for the user and we're on the browser,
+    // try to get the browser location
+    if (
+      !(currentUserLat && currentUserLng) &&
+      !isServer &&
+      typeof window !== 'undefined' &&
+      typeof navigator !== 'undefined' &&
+      navigator &&
+      navigator.geolocation
+    ) {
       navigator.geolocation.getCurrentPosition((position) => {
-          if(position && position.coords) {
-            const navigatorLat = position.coords.latitude
-            const navigatorLng = position.coords.longitude
-            if (onLoadFinished)
-              onLoadFinished({lat: navigatorLat, lng: navigatorLng, loading: false, known: true});
-          } else {
-            if (onLoadFinished)
-              onLoadFinished({lat: placeholderLat, lng: placeholderLng, loading: false, known: false});
-          }
-        },
-        (error) => {
-          if (onLoadFinished)
-            onLoadFinished({lat: placeholderLat, lng: placeholderLng, loading: false, known: false});
+        if (position && position.coords) {
+          const navigatorLat = position.coords.latitude
+          const navigatorLng = position.coords.longitude
+          setLocation({lat: navigatorLat, lng: navigatorLng, loading: false, known: true})
+        } else {
+          setLocation({lat: placeholderLat, lng: placeholderLng, loading: false, known: false})
         }
-      );
-      return {lat: placeholderLat, lng:placeholderLng, loading: true, known: false};
+      },
+      (error) => {
+        setLocation({lat: placeholderLat, lng: placeholderLng, loading: false, known: false})
+      }
+    )
     }
-
-    return {lat: placeholderLat, lng:placeholderLng, loading: false, known: false};
-  }
+  }, [currentUserLat, currentUserLng, placeholderLat, placeholderLng])
```

Comment:
Should this have an empty dependency list? If the currentUserLat/Lng changes, do we want this running?again? Will it infinite loop?

Reply:
That was my intention, but I figured I should err on the side of following linting rules. I checked and it doesn't infinite loop (since we never replace the user's location with the browser location if they already have one), but it does get called one extra time every time the user changes the location in the input. So not the best. I'll go ahead and eslint-disable it so we can have it just run once.